atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -552,16 +552,18 @@ private TransactionManager serializableInternal(@Output List<AutoCloseable> clos
         return transactionManager;
     }
 
+    abstract Optional<String> serviceIdentifier();
```

Comment:
nit: can we name this `serviceIdentifierOverride` or `explicitServiceIdentifier` or something like that?